### PR TITLE
add Plausible event - Story-Started and Story-Finished

### PR DESCRIPTION
**Pull request by daria31v**

### CODE REVIEW

**Review by met — APPROVED** [review]
Looks perfect to me! 👍

File: components/hooks/useStoryReader.ts [hunk 1]
```diff
@@ -69,17 +87,31 @@ export const useStoryReader = (id: string) => {
...
     audio.current = new Audio(source);
     audio.current.ontimeupdate = handleTimeUpdate;
     audio.current.onended = handleAudioEnd;
-
     return () => {
       stopStory();
     };
   }, [source, stopStory]);
 
+  useEffect(() => {
```
Comment (jejdacz) [inline components/hooks/useStoryReader.ts:105]: I tested the App a bit. When you finished the story then try to move slider back. Then you get Story-Finished event hundred times. Replace temporarily plausible function with this `const plausible = (event,data) => console.log(event);` and test it . Useffects are trickiest part of react. I would start to look there. When you need to set variable included in condition in Useffect inside same Useffects use const switch = useRef(true); . This variable doesn have to be included in depc so it cant make infinite loop. Try to look at my tracking hook /components/basecomponents/Exercise/utils/useTracking.ts

**Review by jejdacz — CHANGES_REQUESTED** [review]
When story is finished then moving of slider back triggers the Story-finished event hundred times.

**Comment by met** [comment]
> When story is finished then moving of slider back triggers the Story-finished event hundred times.

@jejdacz Did you try last commit in this PR? This bug with repeating Story-Ended I have seen in the beginning but after Daria corrected in last commit it I do not see them again.

File: components/hooks/useStoryReader.ts (hunk 1 above)
Comment (daria31v) [reply components/hooks/useStoryReader.ts:105]: okay, I`ll think about this.
thanks

File: components/hooks/useStoryReader.ts [hunk 2]
```diff
@@ -69,17 +87,31 @@ export const useStoryReader = (id: string) => {
...
+    if (!isFirstEnd) {
+      if (audioEnded) {
+        plausible('Story-Finished', { props: { story_audio_language: languagePlay, story_name: id, language, kiosk } });
+      }
+      return;
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (daria31v) [reply components/hooks/useStoryReader.ts:112]: I limited in this way the condition under which it works useEffect

**Comment by jejdacz** [comment]
> > When story is finished then moving of slider back triggers the Story-finished event hundred times.
> 
> @jejdacz Did you try last commit in this PR? This bug with repeating Story-Ended I have seen in the beginning but after Daria corrected in last commit it I do not see them again.

Im synced to last commit.

File: components/hooks/useStoryReader.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jejdacz) [reply components/hooks/useStoryReader.ts:112]: Deps are not supposed to be used like that.

**Comment by jejdacz** [comment]
Be aware that code needs to stay functional when language is switched. The first switch is in menu and second is in story itself. When you have some of this variables in useeffect deps then it is triggered. So you need condition inside to watch only relevant variables.

**Comment by jejdacz** [comment]
I will send you my solution to slack.

**Comment by daria31v** [comment]
I ask you to consider this option.
And I want to note that, unfortunately, the option with the variable **const audioEnded = useRef(false);** does not work correctly for me.
Namely, the event is not sent at the right time.
This edition is stable for me.
Please test.

**Review by jejdacz — APPROVED** [review]
Now it  works, nice. Please delete the commented code and it's done.

File: components/hooks/useStoryReader.ts (hunk 1 above)
Comment (jejdacz) [reply components/hooks/useStoryReader.ts:106]: Good job.

**Pull request merged**